pkg/master/controller.go

```diff
@@ -131,11 +131,13 @@ func (c *completedConfig) NewBootstrapController(legacyRESTStorage corerest.Lega
 	}
 }
 
+// PostStartHook only runs Start currently
 func (c *Controller) PostStartHook(hookContext genericapiserver.PostStartHookContext) error {
 	c.Start()
 	return nil
 }
 
+// PreShutdownHook only runs Stop currently
```

Comment:
PreShutdownHook triggers the actions needed to shut down the API Server cleanly.

Reply:
done